salt/modules/file.py

```diff
@@ -2393,14 +2394,22 @@ def blockreplace(path,
                         # end of block detected
                         in_block = False
 
-                        # Check for multi-line '\n' terminated content as split will
-                        # introduce an unwanted additional new line.
-                        if content and content[-1] == '\n':
-                            content = content[:-1]
+                        # Separate the content into lines. Account for Windows
+                        # style line endings using os.linesep, then by linux
+                        # style line endings
+                        split_content = []
+                        for linesep_line in content.split(os.linesep):
+                            for content_line in linesep_line.split('\n'):
```

Comment:
Can you explain the double loop here? Assuming that `content` here is a string, wouldn't `os.linesep` "just work" here? Or are you simply considering the case that we may be dealing with files containing unix line separators on a Windows box and vice-versa?

Reply:
Yes... I was considering cases where a file may contain both line separators. I thought to split on the larger first (`\r\n`) and then on the smaller one (`\n`). I couldn't think of any instances where `\r` might be used.

Reply:
Yeah, that is a good call. It's probably a good idea to document that's the reason you're doing it this way then, by rewording the comment above. That way it'll be easier to understand for any community member who may be working with this code in the future.

Reply:
Done